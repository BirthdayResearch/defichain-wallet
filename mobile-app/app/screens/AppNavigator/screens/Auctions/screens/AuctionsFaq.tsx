import { ThemedScrollView, ThemedText } from '@components/themed'
import { AccordionContent, WalletAccordion } from '@components/WalletAccordion'
import { tailwind } from '@tailwind'
import { translate } from '@translations'

export function AuctionsFaq (): JSX.Element {
  const faqContent: AccordionContent[] = [
    {
      title: translate('components/AuctionsFaq', 'How is the minimum starting bid determined?'),
      content: [{
        text: translate('components/AuctionsFaq', 'The minimum starting bid is 105% of the batch\'s outstanding loan value. The additional 5% is included to cover the liquidation penalty, which is incurred upon vault liquidation.'),
        type: 'paragraph'
      }]
    },
    {
      title: translate('components/AuctionsFaq', 'How is the minimum next bid determined?'),
      content: [{
        text: translate('components/AuctionsFaq', 'The minimum next bid is 1% above the previous bid price.'),
        type: 'paragraph'
      }]
    }
  ]

  return (
    <ThemedScrollView
      contentContainerStyle={tailwind('p-6 pb-8')}
      testID='auctions_faq'
    >
      <ThemedText
        style={tailwind('text-lg font-semibold')}
      >
        {translate('components/AuctionsFaq', 'Auctions')}
      </ThemedText>

      <ThemedText
        style={tailwind('mt-2 text-sm')}
      >
        {translate('components/AuctionsFaq', 'Auction is an activity where users bid on the collateral in liquidated vaults to recover the vaults\' outstanding loans. Liquidation occurs when a vault\'s collateralization ratio falls below its minimum requirement, upon which the vault collateral will be split into batches of $10,000 USD for auction.')}
      </ThemedText>

      <WalletAccordion
        testID='auctions_faq_accordion'
        title={translate('components/AuctionsFaq', 'FREQUENTLY ASKED QUESTIONS')}
        content={faqContent}
      />
    </ThemedScrollView>
  )
}
