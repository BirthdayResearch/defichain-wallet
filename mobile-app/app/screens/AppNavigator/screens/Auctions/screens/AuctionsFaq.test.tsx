import { render } from '@testing-library/react-native'

import { AuctionsFaq } from './AuctionsFaq'

jest.mock('@shared-contexts/ThemeProvider')

describe('Auctions FAQ screen', () => {
  it('should match snapshot', async () => {
    const rendered = render(<AuctionsFaq />)
    expect(rendered.toJSON()).toMatchSnapshot()
  })
})
