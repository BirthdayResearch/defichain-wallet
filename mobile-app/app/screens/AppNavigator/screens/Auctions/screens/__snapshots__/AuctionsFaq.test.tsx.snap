// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auctions FAQ screen should match snapshot 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "paddingBottom": 32,
      "paddingLeft": 24,
      "paddingRight": 24,
      "paddingTop": 24,
    }
  }
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "rgba(245, 245, 245, 1)",
      },
    ]
  }
  testID="auctions_faq"
>
  <View>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "RegularFont",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
          Array [
            Object {
              "fontFamily": "SemiBoldFont",
              "fontSize": 18,
              "fontWeight": "600",
              "lineHeight": 28,
            },
            Object {
              "color": "rgba(0, 0, 0, 1)",
            },
          ],
        ]
      }
    >
      Auctions
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "RegularFont",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
          Array [
            Object {
              "fontSize": 14,
              "lineHeight": 20,
              "marginTop": 8,
            },
            Object {
              "color": "rgba(0, 0, 0, 1)",
            },
          ],
        ]
      }
    >
      Auction is an activity where users bid on the collateral in liquidated vaults to recover the vaults' outstanding loans. Liquidation occurs when a vault's collateralization ratio falls below its minimum requirement, upon which the vault collateral will be split into batches of $10,000 USD for auction.
    </Text>
    <RCTScrollView
      style={
        Array [
          undefined,
          Object {
            "backgroundColor": "rgba(245, 245, 245, 1)",
          },
        ]
      }
      testID="auctions_faq_accordion"
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "RegularFont",
                "fontSize": 16,
                "fontWeight": "400",
                "lineHeight": 24,
              },
              Array [
                Object {
                  "fontFamily": "MediumFont",
                  "fontSize": 12,
                  "fontWeight": "500",
                  "lineHeight": 16,
                  "marginTop": 32,
                },
                Object {
                  "color": "rgba(163, 163, 163, 1)",
                },
              ],
            ]
          }
          text="FREQUENTLY ASKED QUESTIONS"
        >
          FREQUENTLY ASKED QUESTIONS
        </Text>
        <View
          style={
            Array [
              Object {
                "borderBottomLeftRadius": 8,
                "borderBottomRightRadius": 8,
                "borderBottomWidth": 1,
                "borderLeftWidth": 1,
                "borderRightWidth": 1,
                "borderTopLeftRadius": 8,
                "borderTopRightRadius": 8,
                "borderTopWidth": 1,
                "marginTop": 8,
                "overflow": "hidden",
              },
              Object {
                "backgroundColor": "rgba(255, 255, 255, 1)",
                "borderBottomColor": "rgba(229, 229, 229, 1)",
                "borderLeftColor": "rgba(229, 229, 229, 1)",
                "borderRightColor": "rgba(229, 229, 229, 1)",
                "borderTopColor": "rgba(229, 229, 229, 1)",
              },
            ]
          }
        >
          <View
            style={Object {}}
          >
            <View
              accessibilityState={
                Object {
                  "expanded": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <View
                style={
                  Array [
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                        "paddingBottom": 16,
                        "paddingLeft": 16,
                        "paddingRight": 16,
                        "paddingTop": 16,
                      },
                      Object {
                        "borderBottomWidth": 1,
                      },
                      false,
                    ],
                    Object {
                      "borderBottomColor": "rgba(229, 229, 229, 1)",
                      "borderLeftColor": "rgba(229, 229, 229, 1)",
                      "borderRightColor": "rgba(229, 229, 229, 1)",
                      "borderTopColor": "rgba(229, 229, 229, 1)",
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "RegularFont",
                        "fontSize": 16,
                        "fontWeight": "400",
                        "lineHeight": 24,
                      },
                      Array [
                        Array [
                          Object {
                            "flexBasis": "0%",
                            "flexGrow": 1,
                            "flexShrink": 1,
                            "fontSize": 14,
                            "lineHeight": 20,
                          },
                          false,
                        ],
                        Object {
                          "color": "rgba(115, 115, 115, 1)",
                        },
                      ],
                    ]
                  }
                >
                  How is the minimum starting bid determined?
                </Text>
                <Text />
              </View>
            </View>
            <View
              pointerEvents="none"
              style={
                Object {
                  "height": 0,
                  "overflow": "hidden",
                }
              }
            >
              <View
                forwardedRef={[Function]}
                onLayout={[Function]}
                style={Object {}}
              >
                <View
                  style={
                    Array [
                      Array [
                        Object {
                          "paddingBottom": 16,
                          "paddingLeft": 16,
                          "paddingRight": 16,
                          "paddingTop": 0,
                        },
                        false,
                      ],
                      Object {
                        "borderBottomColor": "rgba(229, 229, 229, 1)",
                        "borderLeftColor": "rgba(229, 229, 229, 1)",
                        "borderRightColor": "rgba(229, 229, 229, 1)",
                        "borderTopColor": "rgba(229, 229, 229, 1)",
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                        "justifyContent": "flex-start",
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "fontFamily": "RegularFont",
                            "fontSize": 16,
                            "fontWeight": "400",
                            "lineHeight": 24,
                          },
                          Array [
                            Object {
                              "flexBasis": "0%",
                              "flexGrow": 1,
                              "flexShrink": 1,
                              "fontSize": 14,
                              "lineHeight": 20,
                            },
                            Object {
                              "color": "rgba(82, 82, 82, 1)",
                            },
                          ],
                        ]
                      }
                    >
                      The minimum starting bid is 105% of the batch's outstanding loan value. The additional 5% is included to cover the liquidation penalty, which is incurred upon vault liquidation.
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={Object {}}
          >
            <View
              accessibilityState={
                Object {
                  "expanded": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <View
                style={
                  Array [
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                        "paddingBottom": 16,
                        "paddingLeft": 16,
                        "paddingRight": 16,
                        "paddingTop": 16,
                      },
                      false,
                      false,
                    ],
                    Object {
                      "borderBottomColor": "rgba(229, 229, 229, 1)",
                      "borderLeftColor": "rgba(229, 229, 229, 1)",
                      "borderRightColor": "rgba(229, 229, 229, 1)",
                      "borderTopColor": "rgba(229, 229, 229, 1)",
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "RegularFont",
                        "fontSize": 16,
                        "fontWeight": "400",
                        "lineHeight": 24,
                      },
                      Array [
                        Array [
                          Object {
                            "flexBasis": "0%",
                            "flexGrow": 1,
                            "flexShrink": 1,
                            "fontSize": 14,
                            "lineHeight": 20,
                          },
                          false,
                        ],
                        Object {
                          "color": "rgba(115, 115, 115, 1)",
                        },
                      ],
                    ]
                  }
                >
                  How is the minimum next bid determined?
                </Text>
                <Text />
              </View>
            </View>
            <View
              pointerEvents="none"
              style={
                Object {
                  "height": 0,
                  "overflow": "hidden",
                }
              }
            >
              <View
                forwardedRef={[Function]}
                onLayout={[Function]}
                style={Object {}}
              >
                <View
                  style={
                    Array [
                      Array [
                        Object {
                          "paddingBottom": 16,
                          "paddingLeft": 16,
                          "paddingRight": 16,
                          "paddingTop": 0,
                        },
                        false,
                      ],
                      Object {
                        "borderBottomColor": "rgba(229, 229, 229, 1)",
                        "borderLeftColor": "rgba(229, 229, 229, 1)",
                        "borderRightColor": "rgba(229, 229, 229, 1)",
                        "borderTopColor": "rgba(229, 229, 229, 1)",
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                        "justifyContent": "flex-start",
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "fontFamily": "RegularFont",
                            "fontSize": 16,
                            "fontWeight": "400",
                            "lineHeight": 24,
                          },
                          Array [
                            Object {
                              "flexBasis": "0%",
                              "flexGrow": 1,
                              "flexShrink": 1,
                              "fontSize": 14,
                              "lineHeight": 20,
                            },
                            Object {
                              "color": "rgba(82, 82, 82, 1)",
                            },
                          ],
                        ]
                      }
                    >
                      The minimum next bid is 1% above the previous bid price.
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTScrollView>
`;
